wren-main/src/main/java/io/wren/main/sql/snowflake/RewriteCast.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.wren.main.sql.snowflake;
+
+import io.trino.sql.tree.Cast;
+import io.trino.sql.tree.DataType;
+import io.trino.sql.tree.Expression;
+import io.trino.sql.tree.FunctionCall;
+import io.trino.sql.tree.GenericDataType;
+import io.trino.sql.tree.Node;
+import io.trino.sql.tree.QualifiedName;
+import io.trino.sql.tree.StringLiteral;
+import io.trino.sql.tree.TypeParameter;
+import io.wren.base.sqlrewrite.BaseRewriter;
+import io.wren.main.metadata.Metadata;
+import io.wren.main.sql.SqlRewrite;
+
+import java.util.List;
+import java.util.Optional;
+
+public class RewriteCast
+        implements SqlRewrite
+{
+    public static final RewriteCast INSTANCE = new RewriteCast();
+
+    private RewriteCast() {}
+
+    @Override
+    public Node rewrite(Node node, Metadata metadata)
+    {
+        return new CastRewriter().process(node, null);
+    }
+
+    private static class CastRewriter
+            extends BaseRewriter<Void>
+    {
+        private boolean targetDataTypeIsBytea;
```

Comment:
We can't set the flag to be a global variable because the child of a cast could be not only `StringLiteral` but others, such as a function call. Consider a case:
```
SELECT CAST(reverse('123') AS bytea) c1
```
If we use a global variable to handle this flag, the result will be
```
SELECT CAST(reverse(HEX_ENCODE('123')) AS bytea) c1
```
However, the expected result is 
```
SELECT CAST(HEX_ENCODE(reverse('123')) AS bytea) c1
```